### DISCUSSION TITLE
Access denied from Iran

### DISCUSSION THREAD

**Discussion by esmaeilsg1** [community/community#47732] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I am from Iran and I get this error:
activation failed: "At this time, Copilot is not available in your location"

But I read this in Site Policy:
https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls.

As a result of our advocacy and hard work with U.S. regulators, GitHub has secured a license from the U.S. Treasury Department’s Office of Foreign Assets Control (OFAC) to restore our cloud services to developers in Iran. And we will continue to work with U.S. regulators to secure similar licenses to offer collaboration services to developers in other sanctioned regions, such as Syria and Crimea. We are committed to advocating for the broadest possible developer access, as we believe offering code collaboration services for developers in sanctioned regions makes the global developer community stronger, advances human progress, and supports the enduring U.S. foreign policy of promoting free speech and the free flow of information. For information about the license we secured for developers in Iran, read [the blog post](https://github.blog/2021-01-05-advancing-developer-freedom-github-is-fully-available-in-iran).

**Comment by Akash1134** [comment]
Hi there,

Sorry for the difficulty you're experiencing. I'm afraid that due to current trade controls, your GitHub account cannot make use of Copilot.

While GitHub obtained a license in 2021 to provide cloud services to Iranian users, Copilot is not included in the scope of this license.

We want every developer to be able to collaborate on GitHub, and are undertaking efforts to increase Copilot availability around the world, so please know that GitHub will continue to advocate for the right of developers in trade-restricted regions to access tools for code collaboration and communication.

For more details, see the page you've linked above [GitHub and Trade Controls](https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls#can-you-clarify-availability-of-github-to-cuban-developers)

**Comment by parham2013** [comment]
you can use vpn to access copilot but remember copilot is not free,
and programs like pycharm won't use v2rayn directly so you need to change settings a bit,
first run v2rayn program and connect to a vpn,
if you use pycharm go in setting and do this:
if your port number is different you can see it in your v2rayn folder config.json file, where 'tag' is 'http'
![Untitled](https://user-images.githubusercontent.com/74326920/236598243-346bd36e-eacf-4713-8359-7149b1deda55.png)

**Comment by joniadaniel** [comment]
The protests, which were sparked on 16 September after the death of a 22-year-old Kurdish woman in police custody, show no sign of subsiding. On Thursday, protesters torched police stations and vehicles in several cities.
regards: <a href="https://greenratcontrol.com/how-to-kill-mosquito-larvae-in-pools/">mosquito larvae in pools</a>
